src/plot_api/plot_api.js

```diff
@@ -2219,7 +2219,7 @@ Plotly.animate = function(gd, frameOrGroupNameOrFrameList, animationOpts) {
 
                 var nextFrame = {
                     frame: computedFrame,
-                    name: frameList[i].name,
+                    name: frameList[i].name || computedFrame.name,
```

Comment:
Good catch. Moved. I added an extra container for frames internally so that I'm not cramming any extra properties together that could conceivably conflict.